x-pack/plugins/enterprise_search/server/utils/elasticsearch_error_handler.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IKibanaResponse, RequestHandler } from '@kbn/core/server';
+import { Logger } from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+
+import { ErrorCode } from '../../common/types/error_codes';
+
+import { createError } from './create_error';
+
+export function elasticsearchErrorHandler(
+  log: Logger,
+  requestHandler: RequestHandler
+): RequestHandler {
```

Comment:
the `request` argument is losing its typing while being passed through.  This causes typescript to throw errors because it doesn't think it has access to specific properties of the request, an example is needing `request.body.index_name` https://github.com/elastic/kibana/pull/138341/files#diff-08d953b6bbfdec84ec8002022f1fa3d7b1c7c6fe30fdc781106d8898059d22f9R40